_posts/2021/2021-09-16-5-things-i-learned-in-10-years.md

```diff
@@ -0,0 +1,124 @@
+- You stay interested in learning new things
+- You can break the ice easier with new colleagues
+- As you become more and more familiar with concepts outside of your knowledge, you become slowly immune against the [Paradox of Expertise](https://www.psychologytoday.com/us/blog/tracking-wonder/201906/the-paradox-expertise){:target="_blank"}
+- You will get tasks with more responsibility from your employer
+
+
+
+### 4. Learn software architecture (how to integrate external systems: file transfer, remote procedure, messaging)
+
+A lot of developers love to code.
+Design patterns will pop up along the developer journey for sure.
+Patterns like MVC, Strategy, Bridge, Adapter, Singleton will be known for many.
+But there is a lot more to software architecture than this.
+For example, in modern software architecture, you have to integrate a lot of external services instead of implementing them by yourself.
+There are lots of ways to do this (e.g. Webservice based solutions, file-based integrations, and more can be found in this [book](https://www.amazon.de/Patterns-Enterprise-Application-Architecture-Addison-Wesley-ebook/dp/B008OHVDFM/ref=rvi_1/){:target="_blank"}).
+Choosing the right (also technical) solution or pattern can base on many factors.
+One of the biggest influences you have on your decision is based on the purpose and size of the software.
+There are huge differences between B2B, B2C, B2G or C2C applications.
+Each will bring their own rules and pitfalls.
+This could be:
+
+- traffic load (B2C)
+- high complexity (B2B)
+- obligation of special laws (B2G)
+- transaction/distribution strategy (C2C)
+
+Documentation of decisions becomes very important and is one pillar of an effective and robust architecture for the distant future.
+If you document your decisions (including arguments for/against other options as well as assumptions) they will make future decisions and your system more robust.
+
+### 5. Be open-minded
+
+While working for several companies over all those years, there is one thing you will always stumble upon - Discussions and disagreements.
+Every developer has a lot of knowledge and thinks his/her decision or opinion is the right one.
+And the developer tries not to give in as pride is standing in its way.
+
+I would lie if this would not have happened to me.
+At some point, I read an article not just about being open-minded, but rather how to be.
+If I need to break it down into short and concise bullet points, I would describe it as:
+
+- Always approach disagreements with curiosity
+- Always fearing being wrong, no matter how confident I think that I’m right
+- Try to fully *understand* (not just *hear*) other peoples arguments first, and then reject or accept those arguments.
+Furthermore, try to understand *what* people are saying, and their reasoning behind it.
+- Look for the best answer, not the best answer you can come up with on your own but rather as a team
```

Comment:
"Look for the best answer, not the best answer you can come up with by yourself, but as a team."